File: 0169-majority-element/0169-majority-element.py
class Solution(object):
    def majorityElement(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        num_array = {}

        for num in nums:
            if num in num_array:
                num_array[num] += 1
            else:
                num_array[num] = 1

        length = len(nums)/2

        for key, val in num_array.items():
            if val > length:
                return key
        

File: 0169-majority-element/0169-majority-element.cpp
// Include the map library
#include <map>
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        unordered_map<int, int> store;
        for (int x: nums){
            store[x]++;
            if(store[x] > nums.size() / 2) return x;
        }
        return -1;
    }
};